opm/autodiff/FlowMain.hpp

```diff
@@ -101,6 +101,33 @@
 namespace Opm
 {
 
+    boost::filesystem::path simulationCaseName( const std::string& casename ) {
+        namespace fs = boost::filesystem;
+
+        const auto exists = []( const fs::path& f ) {
```

Comment:
This assumes C++14.  In C++11 you can omit the return type only if the body of the closure consists of a [single `return` statement](http://en.cppreference.com/w/cpp/language/lambda).